sdks/python/apache_beam/transforms/external.py

```diff
@@ -443,7 +443,8 @@ def _normalize(coder_proto):
           self._external_namespace), (id, self._external_namespace)
       new_proto = beam_runner_api_pb2.PTransform(
           unique_name=proto.unique_name,
-          spec=proto.spec,
+          # If URN is not set this is an empty spec.
+          spec=proto.spec if proto.spec.urn else None,
```

Comment:
Should this also check `proto.spec`?
```suggestion
          spec=proto.spec if proto.spec and proto.spec.urn else None,
```

Reply:
Unfortunately with Proto3 you cannot really check whether a field is set or not it seems and Python will give a non zero byte value even for an empty "spec" field. You can just check the values of the fields.